app/javascript/controllers/soft_validations_controller.js

```diff
@@ -0,0 +1,28 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = [ "override" ]
+
+  initialize() {
+    let url = window.location.href + "/soft_validations"
+    let div = this.overrideTarget
+    fetch(url, { headers: { accept: "application/json" } })
+      .then(response => response.json())
+      .then((response) => {
+        if(response["show"]){
+          div.style.display = "block"
+          let innerHTML = div.innerHTML
+          innerHTML = innerHTML.replace("soft-validations-placeholder-message", response["label"])
+          innerHTML = innerHTML.replace("soft-validations-placeholder-hint-text", response["hint"])
+          div.innerHTML = innerHTML
+        } else {
+          div.style.display = "none"
+          let buttons = document.getElementsByName(`case_log[override_net_income_validation][]`)
+          Object.entries(buttons).forEach(([idx, button]) => {
+            button.checked = false
+          })
```

Comment:
I'm assuming this is covered by the existing case log feature specs and the new one you added so there's no need to unit test this?

Reply:
Exactly, this is just a refactoring of approach, not a change of outcomes. It does solve a bug with the back link but there is a test for that added.

Reply:
using a .map here may. be quicker than a forEach